omod/src/test/java/org/openmrs/module/fhir2/providers/r3/ConditionFhirR3ResourceProviderWebTest.java

```diff
@@ -220,222 +232,289 @@ public void shouldCreateNewConditionGivenValidConditionResource() throws Excepti
 			Objects.requireNonNull(is);
 			conditionJson = IOUtils.toString(is, StandardCharsets.UTF_8);
 		}
-		
+
 		when(conditionService.saveCondition(any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(condition);
-		
-		MockHttpServletResponse response = post("/Condition").jsonContent(conditionJson).accept(FhirMediaTypes.JSON).go();
-		
+
+		MockHttpServletResponse response = post("/Condition").jsonContent(conditionJson).accept(FhirMediaTypes.JSON)
+				.go();
+
 		assertThat(response, isCreated());
 		assertThat(response.getStatus(), is(201));
 	}
+
+	@Test
+	public void updateCondition_shouldUpdateExistingCondition() throws Exception {
+		String conditionJson;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_CONDITION_PATH)) {
+			Objects.requireNonNull(is);
+			conditionJson = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		when(conditionService.update(anyString(), any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(conditionJson);
```

Comment:
```suggestion

		when(conditionService.update(anyString(), any(org.hl7.fhir.r4.model.Condition.class))).thenReturn(conditionJson);
```

Reply:
Also, note that this line is failing because it needs to return a Condition object and not a String.

Reply:
Thanks for the couple of comments, One more thing am requesting am we entitled to add other different condition classes besides these ones?, travis keep failing